python/ray/data/dataset.py

```diff
@@ -2289,7 +2289,7 @@ def __next__(self) -> "Dataset[T]":
                     raise StopIteration
                 self._ds._set_epoch(self._i)
                 self._i += 1
-                return lambda: self._ds
+                return lambda: self._ds.force_reads()
```

Comment:
Is this force read needed if the pipeline force-reads on each stage,?

Reply:
Fixed